src/FakeItEasy/Configuration/FakeConfigurationManager.cs

```diff
@@ -61,11 +62,14 @@ public IAnyCallConfigurationWithNoReturnTypeSpecified CallTo(object fakeObject)
         public IPropertySetterAnyValueConfiguration<TValue> CallToSet<TValue>(Expression<Func<TValue>> propertySpecification)
         {
             Guard.AgainstNull(propertySpecification, nameof(propertySpecification));
-            this.AssertThatMemberCanBeIntercepted(propertySpecification);
+            var parsedCallExpression = this.callExpressionParser.Parse(propertySpecification);
+            this.AssertThatMemberCanBeIntercepted(parsedCallExpression);
 
-            var setterExpression = BuildSetterFromGetter(propertySpecification);
+            var setterExpression = BuildSetterFromGetter(parsedCallExpression, propertySpecification);
 
-            return new PropertySetterConfiguration<TValue>(setterExpression, this.CreateVoidArgumentValidationConfiguration);
+            return new PropertySetterConfiguration<TValue>(
+                setterExpression,
+                lambda => this.CreateVoidArgumentValidationConfiguration(this.callExpressionParser.Parse(lambda)));

```

Comment:
It took me a moment to understand this change... Sometimes I wish C# had function composition like F#... then we could do `this.callExpressionParser.Parse >> this.CreateVoidArgumentValidationConfiguration` instead of having to use a lambda.